app/src/main/java/com/example/habitsmasher/FollowUserDialog.java

```diff
@@ -180,6 +187,20 @@ public void displayErrorMessage(int messageType) {
                 _userToFollow.setError(INVALID_USERNAME_ERROR_MESSAGE);
                 _userToFollow.requestFocus();
                 break;
+            case CANNOT_FOLLOW_YOURSELF_ERROR:
+                _userToFollow.setError(CANNOT_FOLLOW_YOURSELF_MESSAGE);
+                _userToFollow.requestFocus();
+                break;
         }
     }
+
+    /**
+     * This helper method allows us to trigger an action on the dismissal of the dialog box
+     * @param dialog the dialog box to dismiss
+     */
+    @Override
+    public void onDismiss(@NonNull DialogInterface dialog) {
+        super.onDismiss(dialog);
+        _handler.sendEmptyMessage(0);
+    }
```

Comment:
When the dialog box is dismissed, we send an empty message to the handler. This can trigger an action on the other side, where we can act accordingly.